include/openssl/digest.h

```diff
@@ -300,6 +300,24 @@ OPENSSL_EXPORT void EVP_MD_CTX_set_flags(EVP_MD_CTX *ctx, int flags);
 // EVP_MD_nid calls |EVP_MD_type|.
 OPENSSL_EXPORT int EVP_MD_nid(const EVP_MD *md);
 
+// This function was deprecated from OpenSSL in BoringSSL, but was brought back
+// to AWS-LC.
+//
+// If |pctx| is not NULL, |EVP_MD_CTX_set_pkey_ctx| sets |ctx|'s |EVP_PKEY_CTX|
+// reference to |pctx|. The |EVP_PKEY_CTX| object |pctx| needs to have been
+// initialised before associating it with |ctx|. The hash functions associated
+// to |ctx| and |pctx| must be equal. Once |EVP_MD_CTX_set_pkey_ctx| is called,
```

Comment:
It's related to the discussion we were having here: https://github.com/awslabs/aws-lc/pull/309#discussion_r754711430  The actual signing/verifying enforces that the hash must be equal when we're using RSA. An error pops up when it's not aligned. 
I'm skeptical on adding an extra check in this function since it's from Openssl, and we may want to keep most of the behavior the same for backwards compatibility.

Reply:
After some discussion with @torben-hansen, we'll be adding an extra check to enforce the equal hash.